Middleware/StatisticsMiddleware.cs

```diff
@@ -0,0 +1,33 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using System.Diagnostics;
+using System.Threading.Tasks;
+
+namespace Karma.Middleware
+{
+    public class StatisticsMiddleware
+    {
+        private readonly RequestDelegate _next;
+        private readonly ILogger _logger;
+
+        public StatisticsMiddleware(RequestDelegate next, ILogger<StatisticsMiddleware> logger)
+        {
+            _next = next;
+            _logger = logger;
+        }
+
+        public async Task InvokeAsync(HttpContext context)
+        {
+            var timer = new Stopwatch();
+            timer.Start();
+            context.Response.OnStarting(() => {
+                timer.Stop();
+                var elapsedMs = timer.ElapsedMilliseconds;
+                _logger.LogInformation($"Request for {context.Request.Path} received ({context.Request.ContentLength ?? 0} bytes), response took {elapsedMs} ms");
+                return Task.CompletedTask;
+            });
+
+            await _next(context);
+        }
```

Comment:
Shouldn't await _next(context); be between timer.Start() and timer.Stop()?
Like this (example from lecturer):
![image](https://user-images.githubusercontent.com/90030313/145366326-7dac82bf-4f21-4d9f-bb3c-61803f359ae5.png)

Reply:
Changed to be between the rest of pipeline, think the outcome is the same though since timer stops when response is crafted either way.